dask/array/core.py

```diff
@@ -3492,7 +3492,11 @@ def concatenate(seq, axis=0, allow_unknown_chunksizes=False):
     if not seq:
         raise ValueError("Need array(s) to concatenate")
 
-    meta = np.concatenate([meta_from_array(s) for s in seq], axis=axis)
+    seq_metas = [meta_from_array(s) for s in seq]
+    _concatenate = concatenate_lookup.dispatch(
+        type(max(seq_metas, key=lambda x: getattr(x, "__array_priority__", 0)))
```

Comment:
Do we do similar things with `__array_priority__` in other places where we use concat?

Reply:
Yeah I actually took it from some concatenation code for creating a concrete result from a Dask Array above

https://github.com/dask/dask/blob/4d7e674189ed2473ea2dcc2ddf3536e519bd1da5/dask/array/core.py#L298-L300

Reply:
Great, thanks.